diff --git a/node.gypi b/node.gypi
index f165747..091447e 100644
--- node.gypi
+++ node.gypi
@@ -357,6 +357,18 @@
                       #'-Wl,--no-whole-archive'
                      ]
     }],
+    ['OS=="linux" and target_arch=="arm"', {
+          'cflags' : [
+		    '-nostdinc++',
+            '-isystem../../buildtools/third_party/libc++/trunk/include',
+            '-isystem../../buildtools/third_party/libc++abi/trunk/include'
+          ],
+          'ldflags': [
+		    '../nw/obj/buildtools/third_party/libc++/libc++/*.o',
+            '../nw/obj/buildtools/third_party/libc++abi/libc++abi/*.o',
+            '../nw/obj/buildtools/third_party/libunwind/libunwind/*.o'
+          ]
+    }],
     [ 'OS=="mac" and component == "shared_library"', {
       'xcode_settings': {
         'OTHER_LDFLAGS': [
diff --git a/deps/uv/uv.gyp b/deps/uv/uv.gyp
index 1e63286..55527e0 100644
--- deps/uv/uv.gyp
+++ deps/uv/uv.gyp
@@ -219,6 +219,9 @@
             '-Wno-varargs',
           ],
         }],
+        ['OS=="linux" and target_arch=="arm"', {
+          'cflags': ['--std=gnu99']
+        }],
         [ 'OS in "mac ios"', {
           'sources': [
             'src/unix/darwin.c',
diff --git a/deps/nghttp2/nghttp2.gyp b/deps/nghttp2/nghttp2.gyp
index 06c6023..d3caefd 100644
--- deps/nghttp2/nghttp2.gyp
+++ deps/nghttp2/nghttp2.gyp
@@ -23,6 +23,9 @@
             },
           },
         }],
+        ['target_arch=="arm"', {
+          'defines': ['HAVE_NETINET_IN_H'],
+        }],
         ['0 == 1', {
           'defines': [ 'DEBUGBUILD=1' ]
         }]
diff --git a/deps/openssl/openssl.gyp b/deps/openssl/openssl.gyp
index ef8b740..d7893d6 100644
--- deps/openssl/openssl.gyp
+++ deps/openssl/openssl.gyp
@@ -14,6 +14,7 @@
     'node_byteorder%': 'little',
     'conditions': [
       ['OS=="mac"', { 'openssl_no_asm%': 1 } ],
+      ['OS=="linux" and target_arch=="arm"', { 'openssl_no_asm%': 1 } ],
     ],
   },
   'targets': [
