diff --git a/deps/nghttp2/nghttp2.gyp b/deps/nghttp2/nghttp2.gyp
index ad9ea68..db3aba0 100644
--- deps/nghttp2/nghttp2.gyp
+++ deps/nghttp2/nghttp2.gyp
@@ -26,6 +26,9 @@
             },
           },
         }],
+        ['target_arch=="arm"', {
+          'defines': ['HAVE_NETINET_IN_H'],
+        }],
         ['0 == 1', {
           'defines': [ 'DEBUGBUILD=1' ]
         }]
diff --git a/deps/openssl/openssl.gyp b/deps/openssl/openssl.gyp
index cf136b7..9b3c580 100644
--- deps/openssl/openssl.gyp
+++ deps/openssl/openssl.gyp
@@ -6,6 +6,7 @@
     'node_byteorder%': 'little',
     'conditions': [
       ['OS=="mac"', { 'openssl_no_asm%': 1 } ],
+      ['OS=="linux" and target_arch=="arm"', { 'openssl_no_asm%': 1 } ],
     ],
   },
   'targets': [
diff --git a/deps/openssl/openssl_common.gypi b/deps/openssl/openssl_common.gypi
index ef6b107..c48c5a1 100644
--- deps/openssl/openssl_common.gypi
+++ deps/openssl/openssl_common.gypi
@@ -59,8 +59,7 @@
     }, {
       # linux and others
       'cflags': ['-Wno-missing-field-initializers',
-                 ## TODO: check gcc_version>=4.3
-                 '-Wno-old-style-declaration'],
+                ],
       'defines': [
         'OPENSSLDIR="/etc/ssl"',
         'ENGINESDIR="/dev/null"',
diff --git a/deps/uv/uv.gyp b/deps/uv/uv.gyp
index a56fea7..b874410 100644
--- deps/uv/uv.gyp
+++ deps/uv/uv.gyp
@@ -218,6 +218,9 @@
             '-Wno-varargs',
           ],
         }],
+        ['OS=="linux" and target_arch=="arm"', {
+          'cflags': ['--std=gnu99']
+        }],
         [ 'OS in "mac ios"', {
           'sources': [
             'src/unix/darwin.c',
diff --git a/node.gypi b/node.gypi
index 1ecf51d..6162567 100644
--- node.gypi
+++ node.gypi
@@ -304,6 +304,17 @@
                       #'-Wl,--no-whole-archive'
                      ]
     }],
+    ['OS=="linux" and target_arch=="arm"', {
+          'cflags' : [
+		    '-nostdinc++',
+            '-isystem../../buildtools/third_party/libc++/trunk/include',
+            '-isystem../../buildtools/third_party/libc++abi/trunk/include'
+          ],
+          'ldflags': [
+            '../nw/obj/buildtools/third_party/libc++/libc++/*.o',
+            '../nw/obj/buildtools/third_party/libc++abi/libc++abi/*.o',
+          ]
+    }],
     [ 'OS=="mac" and component == "shared_library"', {
       'xcode_settings': {
         'OTHER_LDFLAGS': [
